### PR TITLE
Download input data from S3/minio 

### PR DESCRIPTION
**Pull request by AzHicham**

### CODE REVIEW

File: server/src/data_downloader.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        Ok(Self {
+            bucket,
+            fusio_data_key: config.data_path_key.clone(),
+            fusio_data_version_id: "".to_string(),
+        })
+    }
+
+    async fn get_file_version_id(&self, file_key: &str) -> Result<String, Error> {
+        let (meta, status_code) = self.bucket.head_object(&file_key).await?;
+        let version_id = if status_code == 200 {
```
Comment (woshilapin) [inline server/src/data_downloader.rs:101]: I'm so much waiting for [this RFC](https://rust-lang.github.io/rfcs/2497-if-let-chains.html) to avoid this multiple nesting and repetition of the `bail!()`. But for now, it's not really easy to do better :shrug:

File: server/src/server_config.rs [hunk 2]
```diff
@@ -179,3 +186,62 @@ impl Default for ChaosParams {
...
+    "eu-west-1".to_string()
+}
+
+pub fn default_bucket_timeout_in_ms() -> u32 {
+    30_000
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum StorageType {
```
Comment (pbench) [inline server/src/server_config.rs:238]:
```suggestion
pub enum DataSource{
```
since this is not really related to "storage" but more "where does my data comes from"

File: launch/src/read.rs [hunk 3]
```diff
@@ -35,44 +35,85 @@
...
     let data = build_transit_data(&base_model);
 
     Ok((data, base_model))
 }
 
-pub fn read_model(launch_params: &LaunchParams) -> Result<BaseModel, Error> {
-    let model = match launch_params.input_data_type {
-        config::InputDataType::Ntfs => transit_model::ntfs::read(&launch_params.input_data_path)?,
+pub fn read_model_from_reader<R>(
+    input_data_reader: R,
```
Comment (pbench) [inline launch/src/read.rs:65]:
```suggestion
pub fn read_model_from_zip_reader<R>(
    input_data_zip_reader: R,
```
since this function expect a reader on a zip file

**Review by pbench — APPROVED** [review]
